frontend/src/js/components/PageViewer/VirtualScroll.tsx

```diff
@@ -131,7 +131,7 @@ export const VirtualScroll: FC<VirtualScrollProps> = ({
 
       setCurrentPages(newCurrentPages);
     }
-  }, [triggerHighlightRefresh, pageCache, currentPages]);
+  }, [triggerHighlightRefresh, pageCache]);
```

Comment:
I think you could wrap `setCurrentPages` in an `if (newCurrentPages !== currentPages)` then you'd avoid the infinite loop but there would still be an extra execution, so maybe fighting the liter is better